pkg/doc/verifiable/jsonld.go

```diff
@@ -294,7 +294,39 @@ func compactJSONLD(doc string, opts *jsonldCredentialOpts, strict bool) error {
 }
 
 func mapsHaveSameStructure(originalMap, compactedMap map[string]interface{}) bool {
-	return reflect.DeepEqual(compactMap(originalMap), compactMap(compactedMap))
+	original := compactMap(originalMap)
+	compacted := compactMap(compactedMap)
+
+	if reflect.DeepEqual(original, compacted) {
+		return true
+	}
+
+	if len(original) != len(compacted) {
+		return false
+	}
+
+	for k, v1 := range original {
+		v1Map, isMap := v1.(map[string]interface{})
+		if isMap {
+			continue
+		}
+
+		v2, present := compacted[k]
+		if !present { // special case - the name of the map was mapped, cannot guess what's a new name
+			continue
+		}
+
+		v2Map, isMap := v2.(map[string]interface{})
+		if !isMap {
+			return false
```

Comment:
This will fail when higher level values are scalar. like id, issuer.

Reply:
Yeah, i made lint happy and introduced one unintentional bug. It should be `!isMap` check at line 310, then we just make sure that we compare two maps